astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -119,8 +119,10 @@ def cirs_to_cirs(from_coo, to_frame):
 @frame_transform_graph.transform(FunctionTransform, ICRS, GCRS)
 def icrs_to_gcrs(icrs_coo, gcrs_frame):
     # first set up the astrometry context for ICRS<->GCRS
-    pv = np.array([gcrs_frame.obsgeoloc.value,
-                   gcrs_frame.obsgeovel.value])
+    pv = np.array([gcrs_frame.obsgeoloc.xyz.value,
+                   gcrs_frame.obsgeovel.xyz.value])
+    if pv.ndim > 2:
+        pv = np.rollaxis(pv, -1, 0)
```

Comment:
I think this will not work for multidimensional locations/velocities: you want axis `0` and `1` both to go to the end, no? I.e., `np.rollaxis(np.rollaxis(pv, 0, pv.ndim), 0, pv.ndim)`

p.s. I was thinking it might be cleaner to just do it in the definition of `pv`, but that is also rather messy
```
loc = gcrs_frame.obsgeoloc.xyz.value
vel = gcrs_frame.obsgeovel.xyz.value
pv = np.stack((np.rollaxis(loc, 0, loc.ndim), np.rollaxis(vel, 0, vel.ndim)), axis=-2)
```

p.s.2 Maybe we should backport `np.moveaxes`...